### DISCUSSION TITLE
Converting images into a gif sequence is appending each image side-by-side as frames

### DISCUSSION THREAD

**Discussion by birchcode** [ImageMagick/ImageMagick#6036] [Help] [answered]
### ImageMagick version

7.1.0-59

### Operating system, version and so on

Osx 11.6, PHP 8.2, Imagick 3.7

### Description

I'm attempting to get all the images in a directory and simply use them as frames as a gif.

```convert -delay 20 -loop 0 *.png animation.gif```

and with PHP

```
            $GIF = new Imagick();
            $GIF->setFormat('gif');

            foreach( $files as $f ) 
            { 
                $frame_filepath = 'frames/' . $f;

                $frame = new Imagick();
                $frame->readImage( $frame_filepath );
                $frame->setImageDelay(41);
                $GIF->addImage($frame);
            }

            
            $GIF->writeImages( 'phpanimation.gif', true );
```

Both of these methods are producing what seems like a buggy result. Instead of each frame being overlayed, they are being appended to the right of the last frame, like so:

![animation](https://user-images.githubusercontent.com/1013599/217068971-7fd09e67-ee5b-474b-8a10-22b0170dbb34.gif)

I should note the source images for the frames are 720x450

Perhaps I need to add something like $GIF-appendImages(false) as maybe it's trying to append by default, but this has no effect. 

Any ideas what could be causing this?

**Comment by fmw42** [comment, marked as answer]
Try adding +repage after reading the PNG files to remove any virtual canvas

`convert -delay 20 *.png +repage  -loop 0 animation.gif`

**Comment by birchcode** [comment]
Thank you!

I found the PHP equivalent is [Imagick::setImagePage](https://www.php.net/manual/en/imagick.setimagepage.php)

```$frame->setImagePage(720,450,0,0);```
